libs/mocks/Cargo.toml

```
@@ -0,0 +1,36 @@
+[package]
+authors = ["Centrifuge <admin@centrifuge.io>"]
+description = "Generic mocks built with mock-builder for traits"
+edition = "2021"
+license = "LGPL-3.0"
+name = "cfg-mocks"
+repository = "https://github.com/centrifuge/centrifuge-chain"
+version = "0.0.1"
+
+[package.metadata.docs.rs]
+targets = ["x86_64-unknown-linux-gnu"]
+
+[dependencies]
+codec = { package = "parity-scale-codec", default-features = false, version = "3.0.0", features = ["derive"] }
+scale-info = { version = "2.3.0", default-features = false, features = ["derive"] }
+frame-support = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.32" }
+frame-system = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.32" }
+sp-std = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.32" }
+
+mock-builder = { path = "../../libs/mock-builder" }
+cfg-primitives = { path = "../primitives", default-features = false }
+cfg-traits = { path = "../traits", default-features = false }
+cfg-types = { path = "../types", default-features = false }
+
+[features]
+default = ["std"]
+std = [
+  "codec/std",
+  "scale-info/std",
+  "frame-support/std",
+  "frame-system/std",
+  "sp-std/std",
+  "cfg-primitives/std",
+  "cfg-traits/std",
+  "cfg-types/std",
+]
```

As a result of #1223, I can claim the `try-runtime` feature is missing here 🙃

```suggestion
]
try-runtime = [
  "frame-support/try-runtime",
  "frame-system/try-runtime",
  "cfg-primitives/try-runtime",
  "cfg-traits/try-runtime",
  "cfg-types/try-runtime",
]

```

---

Good point!

Only a last confirmation, `mocks` should only be used when testing in mocks, under `#[cfg(test)]` feature. Does add `try-runtime` applies in this context?

---

I've finally added them. It does not add damage. Although `cfg-traits` and `cfg-primitives` does not have that feature until #1223 is merged.

---

Thanks! True that. Just hoped #1223 would get merged first since rebasing requires re-approvals from all code-owners 🥲

---

> Only a last confirmation @wischli , `mocks` should only be used when testing in mocks, under `#[cfg(test)]` feature, and added as a `[dev-dependencies]`. Does add `try-runtime` applies in this context? I mean, I think nobody should use this crate with `try-runtime`

My understanding is that any crate which imports `frame-support` as dependency requires adding its `try-runtime` feature and hence the same for any other dependencies which also support `try-runtime`.

---

Thanks for clarify this 👍🏻